eng/versioning/verify_release_set.ps1

```diff
@@ -0,0 +1,97 @@
+param(
+    [Parameter(Mandatory=$true)]
+    [System.String] $ServiceDirectory,
+    # ArtifactsList will be using ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json | Select-Object name, groupId)
+    [Parameter(Mandatory=$true)]
+    [array] $ArtifactsList
+)
+
+$resultsTime = [diagnostics.stopwatch]::StartNew()
+
+# Given the service directory and a list of libraries to be released verify that the list
+# of libraries has full transitive closure. This means that we have to look at each library's
+# pom file's current dependencies and ensure that each dependency is contained within the
+# list to be released.
+$script:FoundError = $false
+$libraryReleaseList = @()
+$missingLibraries = @()
+function Write-Error-With-Color([string]$msg)
+{
+    Write-Host "$($msg)" -ForegroundColor Red
+}
+
+Write-Host "ServiceDirectory=$($ServiceDirectory)"
+Write-Host "ArtifactsList:"
+$ArtifactsList | Format-Table -Property Name, GroupId | Out-String | Write-Host
+foreach($artifact in $ArtifactsList) {
+    $libraryReleaseList += ,($artifact.groupId + ":" + $artifact.name)
```

Comment:
Any reason to not make this a hashset?

Reply:
Six of one, half dozen of the other. I'll updated both libraryReleaseList and missingLibraries to be dictionaries with the next commit.